system-tests/e2e-test-fixtures/src/testFixtures/java/org/eclipse/edc/test/system/utils/TransferTestRunner.java

```diff
@@ -0,0 +1,73 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.test.system.utils;
+
+import jakarta.json.JsonObject;
+import org.eclipse.edc.connector.contract.spi.ContractId;
+import org.eclipse.edc.connector.transfer.spi.types.TransferProcessStates;
+
+import java.time.Duration;
+import java.util.function.UnaryOperator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.ID;
+import static org.eclipse.edc.jsonld.spi.PropertyAndTypeNames.ODRL_POLICY_ATTRIBUTE;
+
+/**
+ * Orchestrate a data transfer between the consumer and the provider and returns the transfer process id.
+ */
+public class TransferTestRunner implements UnaryOperator<String> {
```

Comment:
I'd tend to define custom interfaces instead of using the Java ones, this because with them it's actually impossible to get the usages of the `apply` method.

Reply:
looking better at the usages of this I think it would be better have this as a method of `Participant` with a signature like:
```java
public String requestAsset(String assetId, Participant provider, JsonObject destination, JsonObject privateProperties)
```

because the `apply` is currently always called after object instantiation, a single method would be simpler.